packages/@aws-cdk/aws-iam/lib/identity-base.ts

```diff
@@ -1,16 +1,32 @@
 import { IResource } from '@aws-cdk/core';
 import { IManagedPolicy } from './managed-policy';
-import { Policy } from "./policy";
+import { Policy, PolicyProps } from "./policy";
 import { IPrincipal } from "./principals";
 
 /**
  * A construct that represents an IAM principal, such as a user, group or role.
  */
 export interface IIdentity extends IPrincipal, IResource {
   /**
-   * Attaches an inline policy to this principal.
-   * This is the same as calling `policy.addToXxx(principal)`.
+   * Creates and attaches a Policy to this principal, if it is mutable.
+   * If this principal is immutable, does nothing, and returns undefined.
+   *
+   * @param id the logical identifier to use for the new policy resource.
+   *   It will be created in the scope of this principal (if it's mutable)
+   * @param props the construction properties of the new policy
+   */
+  createAndAttachPolicy(id: string, props?: PolicyProps): Policy | undefined;
```

Comment:
This name is weird... "xAndY" is usually a sign of a bad name. Maybe just `addPolicy`? Semantically it means the same thing: `role.addPolicy('foo', { xxx })`

Reply:
I was also thinking `createAttachedPolicy` might be a better name. What do you think?

For now, changed to `addPolicy`.

Reply:
I'm also worried that `addPolicy` looks to similar to `addToPolicy` (which is another method present on most classes that extend `IIdentity`).